drivers/api/no_os_irq.c

```diff
@@ -63,6 +63,7 @@ int32_t no_os_irq_ctrl_init(struct no_os_irq_ctrl_desc **desc,
 	if (ret)
 		return ret;
 
+	(*desc)->ref++;
```

Comment:
can ref be anything else than 1 and 0 ?

Reply:
Depending on the platform specific implementation, yes. STM32 for example, has only one reference to the IRQ descriptor. If you call init() multiple times, it will use the same pointer which is only allocated once. So in this case `ref` can be anything. I think multiple platforms can benefit from this reference counting mechanism, since they can use only one pointer to an IRQ descriptor.

Reply:
ok makes sense